containers/ecr-viewer/src/app/services/ecrMetadataService.ts

```diff
@@ -96,36 +106,64 @@ export const evaluateEcrMetadata = (fhirBundle: Bundle): EcrMetadata => {
     }
   };
 
-  const fhirERSDWarnings = evaluateAll(
+  const fhirEICRProcessingStatus = evaluateValue(
+    fhirBundle,
+    fhirPathMappings.eICRProcessingStatus,
+  );
+  const fhirEICRProcessingStatusReasonObs = evaluateOne(
     fhirBundle,
-    fhirPathMappings.eRSDwarnings,
+    fhirPathMappings.eICRProcessingStatusReason,
   );
-  const eRSDTextList: ERSDWarning[] = [];
 
-  for (const warning of fhirERSDWarnings) {
-    if (warning.code === "RRVS34") {
-      eRSDTextList.push({
-        warning:
-          "Sending organization is using an malformed eRSD (RCTC) version",
-        versionUsed: "2020-06-23",
-        expectedVersion:
-          "Sending organization should be using one of the following: 2023-10-06, 1.2.2.0, 3.x.x.x.",
-        suggestedSolution:
-          "The trigger code version your organization is using could not be determined. The trigger codes may be out date. Please have your EHR administrators update the version format for complete eCR functioning.",
-      });
-    } else if (warning.code === "RRVS29") {
-      eRSDTextList.push({
-        warning:
-          "Sending organization is using an outdated eRSD (RCTC) version",
-        versionUsed: "2020-06-23",
-        expectedVersion:
-          "Sending organization should be using one of the following: 2023-10-06, 1.2.2.0, 3.x.x.x.",
-        suggestedSolution:
-          "The trigger code version your organization is using is out-of-date. Please have your EHR administration install the current version for complete eCR functioning.",
-      });
+  function geteRSDTextList(
+    processingStatus: string | undefined,
+    reasonObs: Observation | undefined,
+  ): ERSDWarning | JSX.Element {
+    if (processingStatus === "RRVS19") {
+      return eICRProcessingSuccessMsg;
+    } else if (!processingStatus || !reasonObs) {
+      return {};
     }
+
+    const coding = reasonObs.valueCodeableConcept?.coding?.[0];
+    const warningCode = coding?.code;
+    const warningName =
+      coding?.display || eicrProcessingReasonMap[warningCode ?? ""] || "";
```

Comment:
```suggestion
      coding?.display || eicrProcessingReasonMap?.[warningCode] || "";
```

I _think_ this is valid syntax 🤞

Reply:
Hmmm typescript got mad at me, I think bc it's the possibility that `warningCode` might be undefined? Changed it to:
```
    const warningCode = coding?.code ?? "";
    const warningName =
      coding?.display || eicrProcessingReasonMap[warningCode] || "";

```